runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2259,7 +2259,20 @@ remoteCompile(
 
          TR_ASSERT(!metaData->startColdPC, "coldPC should be null");
 
-         compiler->getOptions()->writeLogFileFromServer(logFileStr);
+         int compilationSequenceNumber = compiler->getOptions()->writeLogFileFromServer(logFileStr);
+         if (compilationSequenceNumber)
+            {
+            // if compilationSequenceNumber is not 0, vlog from server is enabled
+            // try to perform an equivalent local compilation and generate vlog
+            // append the same compilationSequenceNumber in the filename
+            compiler->getOptions()->setLogFileForClientOptions(compilationSequenceNumber);
+            if (compiler->compile() != COMPILATION_SUCCEEDED)
+               {
```

Comment:
I do not understand how the local compilation was triggered.
Do you actually see two compilations for this method taking place?

Reply:
`compiler->compile()` I think this command triggers the local compilation, I do see 2 compilation logs generated.

Reply:
In this case, we first open the log file with  `setLogFileForClientOptions`, then triggers local compilation with `compiler->compile()`, in the process trees will be written to the log, after the compilation we close the log.

Reply:
Got it. I missed that compiler->compile().
I didn't think such a solution was possible, but if it works, it works.
I would like this strategy to be done only under some option.